openmc/tallies.py

```diff
@@ -1487,10 +1487,16 @@ def _outer_product(self, other, binary_op):
         other_filters = set(other.filters)
         filter_intersect = self_filters.intersection(other_filters)
 
-        # Align the shared filters to follow in each tally operand
+        # Align the shared filters in successive order
         for i, filter in enumerate(filter_intersect):
-            self_index = self.filters.index(filter)
-            other_filter = other.filters[self_index]
+            self_filter = self.filters[i]
+            other_filter = other.filters[i]
+
+            # If necessary, swap self filter
+            if self_filter != filter:
+                self = self.swap_filters(filter, self_filter)
```

Comment:
Well, you might consider adding an "inplace" parameter to
"swap_filters(...)" that is False by default. If set to True, then it will
not return a new Tally but will instead swap the filters of the existing
tally. This is the notation one sees in a lot of NumPy and Pandas routines
to manipulate ndarrays and dataframes. I considered introducing "inplace"
parameters to some of the tally arithmetic routines at one point, but left
it for another day. Today may be that day :-)
On Nov 19, 2015 6:05 PM, "sam" <notifications@github.com> wrote:

> In openmc/tallies.py
> <https://github.com/mit-crpg/openmc/pull/507#discussion_r45414726>:
>
> >          for i, filter in enumerate(filter_intersect):
> > -            self_index = self.filters.index(filter)
> > -            other_filter = other.filters[self_index]
> > +            self_filter = self.filters[i]
> > +            other_filter = other.filters[i]
> > +
> > +            # If necessary, swap self filter
> > +            if self_filter != filter:
> > +                self = self.swap_filters(filter, self_filter)
>
> I just wanted to swap the filters, but since swap_filters returns a tally
> set self to that returned tally. Do you know if I can just remove self =
> and have the filter swap reflected in self?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/507/files#r45414726>.
>